### PR TITLE
GET-62 Add Explore Occupations page with category list

### PR DESCRIPTION
**Pull request by cwrw**
jira-ticket: https://dfedigital.atlassian.net/browse/GET-62

* add css linter to pipeline
* move check your existing skills out of Job Profile controller into its own controller/views
* add explore occupations controller/views with categories list
* tidy up views and add partials for shared forms

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -1,10 +1,15 @@
 class Category < ApplicationRecord
   has_many :job_profile_categories
   has_many :job_profiles, through: :job_profile_categories, inverse_of: :categories
+  default_scope { order(name: :asc) }
```
Comment (smoothcontract) [inline app/models/category.rb]: I'd really like to avoid adding default scopes - could a scope of `by_name` or something similar be added explicitly here please and then include where needed ?

File: app/views/categories/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<% content_for :breadcrumb do
+    generate_breadcrumbs(
+      t('breadcrumb.job_category'),
+      [
+        [t('breadcrumb.home'), root_path],
+        [t('breadcrumb.explore_occupations'), explore_occupations_path]
+      ]
+    )
+  end
+%>
```
Comment (smoothcontract) [inline app/views/categories/show.html.erb]: This is okay for now but I'm hoping in time we can come up with a better mechanism that doesn't rely on each view having to know which breadcrumbs were followed to reach this page.

File: app/views/explore_occupations/results.html.erb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      <ul class="govuk-list">
+        <% @job_profiles.each do |job_profile| %>
+          <li class="govuk-!-padding-bottom-1">
+            <h3 class="govuk-!-margin-0">
+              <a class="govuk-link no-text-decoration" href="#"><%= job_profile.name %></a>
+            </h3>
+            <p class="govuk-!-margin-0 govuk-!-margin-bottom-3"><%= job_profile.description %></p>
+            <p class="govuk-body-s govuk-!-margin-bottom-1 strong">Salary: £15,500 to £21,000</p>
+            <p class="govuk-body-s muted-text">Found in:
+              <%= job_profile_category_list(job_profile) %>
```
Comment (smoothcontract) [inline app/views/explore_occupations/results.html.erb]: Not sure if it's necessary to add a helper for this, could iterate over the collection of categories in the view similar to the approach used here: https://github.com/DFE-Digital/get-help-to-retrain/pull/25/files#diff-ffb75260f330b885690ab8599cab45f9R22

I think no need to change this, but maybe let's not add more helpers than necessary going forwards? Open to discussion on this too.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -64,6 +64,12 @@ jobs:
       env:
         DOCKER_RUN: $(dockerRun)
 
+    - script: |
```
Comment (smoothcontract) [inline azure-pipelines.yml]: Is this change related to the PR?

File: spec/models/category_spec.rb [hunk 5]
```diff
@@ -42,4 +42,17 @@
...
+
+  describe '.with_job_profiles_without' do
+    it 'returns categories with job profiles without passed in category' do
+      category1 = create(:category, name: 'Administration', job_profiles: [create(:job_profile)])
+      category2 = create(:category, name: 'Education', job_profiles: [create(:job_profile)])
+      expect(described_class.with_job_profiles_without(category1)).to contain_exactly(category2)
+    end
+
+    it 'returns categories with job profiles if category empty' do
+      category = create(:category, name: 'Administration', job_profiles: [create(:job_profile)])
```
Comment (smoothcontract) [inline spec/models/category_spec.rb]: It might be nice to add a trait to the `category` factory e.g. `with_job_profile` to support this. I don't want to use default associations within the factories though as this results in unnecessary models being created for specs that don't need them.

File: app/views/explore_occupations/results.html.erb (hunk 3 above)
Comment (cwrw) [reply app/views/explore_occupations/results.html.erb]: This helper was added to comma delimit the category list, and not just iterating over the categories

File: azure-pipelines.yml (hunk 4 above)
Comment (cwrw) [reply azure-pipelines.yml]: Could extract this to it's own pr, but thought I'd lump it in here as I needed to touch more styles

File: app/views/categories/show.html.erb (hunk 2 above)
Comment (cwrw) [reply app/views/categories/show.html.erb]: I couldn't think of anything past `content_for` and specific links, do you have anything in mind which is cleaner?

File: azure-pipelines.yml (hunk 4 above)
Comment (smoothcontract) [reply azure-pipelines.yml]: Okay, it's cool to include it here.

File: app/views/categories/show.html.erb (hunk 2 above)
Comment (smoothcontract) [reply app/views/categories/show.html.erb]: Nothing concrete yet, let's see how this evolves as we add more pages

File: app/models/category.rb [hunk 6]
```diff
@@ -1,10 +1,15 @@
 class Category < ApplicationRecord
   has_many :job_profile_categories
   has_many :job_profiles, through: :job_profile_categories, inverse_of: :categories
+  scope :by_name, -> { order(name: :asc) }
```
Comment (tavy88) [inline app/models/category.rb]: ORDER ASC is by default.

File: app/models/category.rb [hunk 7]
```diff
@@ -1,10 +1,15 @@
...
 
   def self.with_job_profiles
     includes(:job_profile_categories)
       .where
       .not(job_profile_categories: { job_profile_id: nil })
   end
+
+  def self.with_job_profiles_without(category)
+    Category.with_job_profiles.by_name - [category]
```
Comment (tavy88) [inline app/models/category.rb]: Why not use a `.where.not` clause here to filter out?

File: app/models/category.rb (hunk 6 above)
Comment (cwrw) [reply app/models/category.rb]: its default per id not name
